### ISSUE TITLE
How long did it take to train Prime-agent and Mesh-agent?

### ISSUE THREAD

**Issue by yudopan** [clinplayer/3DModelingRL#1]
Hi Lin,

thanks for sharing your interesting and impressive work!
I am trying to implement it on my own machine.  Could you please tell me how long it took you to train both agents and hardware specifications of the machine you used?

Thanks in advance!

Best,
yudopan

**Comment by clinplayer** [comment]
Hi, for one category, it takes about 4 days for Prim-Agent and about 2 days for Mesh-Agent, on a single GTX2080 card.

**Comment by yudopan** [comment]
thanks so much!

**Comment by yudopan** [comment]
Hi Lin, Happy New Year!
I tried you model on my machine. (RTX Titan and AMD Ryzen Threadripper 3990X). 
For one category, it takes 8 days already to train Prim-Agent and it is still working and it has heavy computation on CPU. I guess something is wrong. For me it took one hour to train one RL Epoch, in default setting there are 2000 RL epochs.
Do you use the default setting in config.py?
And can you share me more details about training?
Thanks in advance!

**Comment by clinplayer** [comment]
Hey, sorry for not making this point clear. The epoch 2000 is just a random number to indicate the maximal training epochs. In practice, I keep observing the reward curve until it converges, and then stop the training manually. The other training details are the same as the file "config.py".

**Comment by jdily** [comment]
Hi,
thx for sharing the source code of this wonderful project.
I have a question following this issue.
I am recently training both Prim and Mesh agents on airplane dataset using the depth image input.
However, it took very long time and I am not sure whether the training is really effective.
So far, for Prim agent, each epoch took 0.85 hours, and I am currently running until 140 epochs (more than 4 days as you mentioned above)
However, the reward curves are not converging based on my observation.
And I compared the currently trained model with the pretrained Prim agent weights (compare the mean rewards using test.py).
The mean reward of the currently trained model is still pretty bad.
I am wondering do you happen to have any ideas on potential mistakes I made here?

thank you
